compiler/src/middle_end/matchcomp.re

```diff
@@ -705,18 +696,123 @@ let rec compile_matrix = (branches, mtx) =>
           "Internal error: convert_match_branches: no non-wildcard column found but not all patterns always match (should be impossible)",
         )
       };
-    let {tree, _} as result = compile_matrix(branches, rotated);
-    {...result, tree: [@implicit_arity] Swap(i, tree)};
+    let result = compile_matrix(rotated);
+    Swap(i, result);
+  };
+
+let prepare_match_branches = branches => {
```

Comment:
So all this function does is massage constant patterns into being guards? Either way, might merit a comment

Reply:
Yup. I can add a comment.